core-bundle/src/Routing/ResponseContext/CoreResponseContextFactory.php

```diff
@@ -26,26 +28,46 @@ public function __construct(ResponseContextAccessor $responseContextAccessor)
         $this->responseContextAccessor = $responseContextAccessor;
     }
 
-    public function createWebpageResponseContext(): WebpageResponseContext
+    public function createResponseContext(): ResponseContext
     {
-        $context = new WebpageResponseContext();
+        $context = new ResponseContext();
+
         $this->responseContextAccessor->setResponseContext($context);
 
         return $context;
     }
 
-    public function createResponseContext(): ResponseContext
+    public function createWebpageResponseContext(): ResponseContext
     {
-        $context = new ResponseContext();
-        $this->responseContextAccessor->setResponseContext($context);
+        $context = $this->createResponseContext();
+        $context->add(new HtmlHeadBag());
 
         return $context;
     }
 
-    public function createContaoWebpageResponseContext(PageModel $pageModel): ContaoWebpageResponseContext
+    public function createContaoWebpageResponseContext(PageModel $pageModel): ResponseContext
     {
-        $context = new ContaoWebpageResponseContext($pageModel);
-        $this->responseContextAccessor->setResponseContext($context);
+        $context = $this->createWebpageResponseContext();
+
+        if (!$context->has(HtmlHeadBag::class)) {
+            return $context;
+        }
+
+        /** @var HtmlHeadBag $htmlHeadBag */
+        $htmlHeadBag = $context->get(HtmlHeadBag::class);
+
+        $title = $pageModel->pageTitle ?: StringUtil::inputEncodedToPlainText($pageModel->title ?: '');
+
+        $htmlHeadBag
+            ->setTitle($title ?: '')
```

Comment:
How can `$title` ever be empty / anything else than an empty string?

Reply:
That's core code, I did not change this.